src/software/thunderscope/common/proto_configuration_widget.py

```diff
@@ -58,9 +70,199 @@ def __init__(
+            )
+
+            self.build_proto(self.proto_to_configure)
+            self.save_current_config_to_file(self.path_to_file)
+            return
+
+        logging.info("Loading protobuf from file: {}".format(path_to_file))
+        with open(path_to_file, "rb") as f:
+            self.proto_to_configure = ThunderbotsConfig()
+            self.proto_to_configure.ParseFromString(f.read())
+
+            self.proto_to_configure.sensor_fusion_config.friendly_color_yellow = (
+                self.is_yellow
+            )
+
+    def send_proto_to_fullsystem(self):
+        """
+        Sending the default configuration protobufs to unix_full_system at startup.
+        Also updates the widget at the same time.
+        """
+        self.update_proto_from_file(self.path_to_file)
+        self.on_change_callback(None, None, self.proto_to_configure)
+        self.update_widget()
+
+    def save_proto_callback(self):
+        """
+        This is a callback for a button that save the current protobuf to disk!
+        """
+        try:
+            save_to_path, should_save = QFileDialog.getSaveFileName(
+                self,
+                "Select Protobufs",
+                self.get_default_savepath(),
+                options=QFileDialog.Option.DontUseNativeDialog,
+            )
+
+            if not should_save:
+                return
+
+
+            self.save_current_config_to_file(save_to_path)
+            self.update_widget()
+        except Exception:
+            logging.warning(
+                "cannot save configuration to {}".format(
+                    save_to_path
+                )
+            )
+
+    def save_current_config_to_file(self, path_to_file):
+        """
+        This save the self.proto_to_configure to path_to_file
+
+        :param path_to_file: the path to file that we are saving to.
+        """
+
+        logging.info("writing to file {}".format(path_to_file))
+        path_to_directory = os.path.dirname(path_to_file)
+        os.makedirs(path_to_directory, exist_ok=True)
+
+        with open(path_to_file, "wb") as f:
+            f.write(self.proto_to_configure.SerializeToString())
+
+    def load_proto_with_file_explorer(self):
+        """
+        loading the current protobuf through file explorer
+        """
+        try:
+            path_to_file, should_open = QFileDialog.getOpenFileName(
+                self,
+                "Select Protobufs",
+                self.get_default_savepath(),
+                options=QFileDialog.Option.DontUseNativeDialog,
+            )
+            if not should_open:
+                return
+
+            self.path_to_file = path_to_file
+            self.update_proto_from_file(path_to_file)
+
+            self.update_widget()
+        except Exception:
+            logging.warning(
+                "cannot load configuration from {}. Are you sure it is a configuration proto?".format(
+                    path_to_file
+                )
+            )
```

Comment:
Could you print the exception as well? Or if there is a common exception, we should handle specifically it in a different `except` clause.

Reply:
done.